From c581749e08ffffb72d5fbd4b22deb4873f428ea6 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Sat, 25 Nov 2017 01:06:37 -0800
Subject: [PATCH] Add workaround for sscanf('NaN')

---
 libdap2/dapcvt.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/libdap2/dapcvt.c b/libdap2/dapcvt.c
index 65cc692..172d514 100644
--- a/libdap2/dapcvt.c
+++ b/libdap2/dapcvt.c
@@ -235,10 +235,22 @@ dapcvtattrval(nc_type etype, void* dst, NClist* src)
 	case NC_FLOAT: {
 	    float* p = (float*)dstmem;
 	    ok = sscanf(s,"%g%n",p,&nread);
+#if defined(_MSC_VER) && (_MSC_VER == 1500)
+	    if (!_strnicmp(s, "NaN", 3)) {
+	      ok = 1;
+	      nread = 3;
+	    }
+#endif
 	    } break;
 	case NC_DOUBLE: {
 	    double* p = (double*)dstmem;
 	    ok = sscanf(s,"%lg%n",p,&nread);
+#if defined(_MSC_VER) && (_MSC_VER == 1500)
+	    if (!_strnicmp(s, "NaN", 3)) {
+	      ok = 1;
+	      nread = 3;
+	    }
+#endif
 	    } break;
 	case NC_UBYTE: {
 	    unsigned char* p = (unsigned char*)dstmem;
-- 
2.10.1

