From 22690ffe1af12ab279a39c764bcca1087c1a13df Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 20 Dec 2017 22:25:48 +0000
Subject: [PATCH 4/8] Add find_package(Threads REQUIRED) to CMakeLists.txt

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7a64d476..a27344bb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -651,6 +651,8 @@ IF(USE_HDF5 OR ENABLE_NETCDF_4)
     SET(HDF5_C_LIBRARY_hdf5 "${HDF5_C_LIBRARY}")
   ENDIF()
 
+  find_package(Threads REQUIRED)
+
   ###
   # The following options are not used in Windows currently.
   ###
-- 
2.14.3 (Apple Git-98)

