From 1bf67f5cc1a382f4774c74137fb05fd34f3db825 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Wed, 22 Nov 2017 14:47:35 -0600
Subject: [PATCH 5/8] Use refactored variable names introduced by 773fe807

---
 libdispatch/utf8proc.h | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/libdispatch/utf8proc.h b/libdispatch/utf8proc.h
index ceea425a..2d29524a 100644
--- a/libdispatch/utf8proc.h
+++ b/libdispatch/utf8proc.h
@@ -81,29 +81,29 @@
 
 #if defined(_MSC_VER) && _MSC_VER < 1800
 // MSVC prior to 2013 lacked stdbool.h and inttypes.h
-typedef signed char utf8proc_int8_t;
-typedef unsigned char utf8proc_uint8_t;
-typedef short utf8proc_int16_t;
-typedef unsigned short utf8proc_uint16_t;
-typedef int utf8proc_int32_t;
-typedef unsigned int utf8proc_uint32_t;
+typedef signed char nc_utf8proc_int8_t;
+typedef unsigned char nc_utf8proc_uint8_t;
+typedef short nc_utf8proc_int16_t;
+typedef unsigned short nc_utf8proc_uint16_t;
+typedef int nc_utf8proc_int32_t;
+typedef unsigned int nc_utf8proc_uint32_t;
 #  ifdef _WIN64
-typedef __int64 utf8proc_ssize_t;
-typedef unsigned __int64 utf8proc_size_t;
+typedef __int64 nc_utf8proc_ssize_t;
+typedef unsigned __int64 nc_utf8proc_size_t;
 #  else
-typedef int utf8proc_ssize_t;
-typedef unsigned int utf8proc_size_t;
+typedef int nc_utf8proc_ssize_t;
+typedef unsigned int nc_utf8proc_size_t;
 #  endif
 #  ifndef __cplusplus
 // emulate C99 bool
-typedef unsigned char utf8proc_bool;
+typedef unsigned char nc_utf8proc_bool;
 #    ifndef __bool_true_false_are_defined
 #      define false 0
 #      define true 1
 #      define __bool_true_false_are_defined 1
 #    endif
 #  else
-typedef bool utf8proc_bool;
+typedef bool nc_utf8proc_bool;
 #  endif
 #else
 #  include <stddef.h>
-- 
2.14.3 (Apple Git-98)

