--- netcdf-c-4.5.0.orig/libdap2/dapcvt.c	2017-10-20 14:20:52.000000000 -0300
+++ netcdf-c-4.5.0/libdap2/dapcvt.c	2017-10-24 11:06:14.777836188 -0300
@@ -217,7 +217,8 @@
 #ifdef _MSC_VER
 	    _ASSERTE(_CrtCheckMemory());
 #endif
-	    if(ival < 0 || ival > NC_MAX_UBYTE) ok = 0;
+	    /* For back compatibility, we allow any value, but force conversion */
+	    ival = (ival & 0xFF);
 	    *p = (char)ival;
 	    } break;
 	case NC_CHAR: {
@@ -246,7 +247,8 @@
 	    unsigned int uval;
 	    ok = sscanf(s,"%u%n",&uval,&nread);
 	    _ASSERTE(_CrtCheckMemory());
-	    if(uval > NC_MAX_UBYTE) ok = 0;
+	    /* For back compatibility, we allow any value, but force conversion */
+	    uval = (uval & 0xFF);
 	    *p = (unsigned char)uval;
 #else
 	    ok = sscanf(s,"%hhu%n",p,&nread);
