--- t_dap3a_new.c       2019-11-21 14:49:32.000000000 +0100
+++ t_dap3a.c   2019-11-21 14:49:45.521775202 +0100
@@ -141,7 +141,7 @@
 
     strlcat(url,"file://",sizeof(url));
     strlcat(url,topsrcdir,sizeof(url));
-    strlcat(url,"/ncdap_test/testdata3/test.02",sizeof(url));
+    strlcat(url,"/testdata3/test.02",sizeof(url));
 
     printf("*** Test: var conversions on URL: %s\n",url);
 
--- test_cvt_new.c      2019-11-21 14:51:16.301257994 +0100
+++ test_cvt.c  2019-11-21 07:56:53.062876537 +0100
@@ -140,7 +140,7 @@
 
     strlcat(url,"file://",sizeof(url));
     strlcat(url,topsrcdir,sizeof(url));
-    strlcat(url,"/testdata3/test.02",sizeof(url));
+    strlcat(url,"/ncdap_test/testdata3/test.02",sizeof(url));
 
     printf("*** Test: var conversions on URL: %s\n",url);

--- test_vara_new.c     2019-11-21 14:51:47.373080970 +0100
+++ test_vara.c 2019-11-21 07:56:53.062876537 +0100
@@ -94,7 +94,7 @@
 
     strncpy(url,"file://",sizeof(url));
     strlcat(url,topsrcdir,sizeof(url));
-    strlcat(url,"/testdata3/test.06",sizeof(url));
+    strlcat(url,"/ncdap_test/testdata3/test.06",sizeof(url));
 
     printf("test_vara: url=%s\n",url);
