diff --git a/dap4_test/baselinethredds/CMakeLists.txt b/dap4_test/baselinethredds/CMakeLists.txt
index 4420c0e..50bc611 100644
--- a/dap4_test/baselinethredds/CMakeLists.txt
+++ b/dap4_test/baselinethredds/CMakeLists.txt
@@ -12,7 +12,9 @@ FILE(COPY ${COPY_FILES} DESTINATION ${CMAKE_CURRENT_BINARY_DIR}/ FILE_PERMISSION
 ##
 # Rename file in support of https://github.com/Unidata/netcdf-c/issues/2077
 ##
+IF(NOT EXISTS ${CMAKE_CURRENT_BINARY_DIR}/GOES16_CONUS_20170821_020218_0.47_1km_33.3N_91.4W.nc4.thredds)
 FILE(RENAME ${CMAKE_CURRENT_BINARY_DIR}/GOES16_TEST1.nc4.thredds ${CMAKE_CURRENT_BINARY_DIR}/GOES16_CONUS_20170821_020218_0.47_1km_33.3N_91.4W.nc4.thredds)
+ENDIF()
 
 FILE(GLOB CUR_EXTRA_DIST RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/*)
 SET(CUR_EXTRA_DIST ${CUR_EXTRA_DIST} CMakeLists.txt)
