diff --git a/CMakeLists.txt b/CMakeLists.txt
index 57467304..32037f1e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -831,11 +831,20 @@ IF(USE_HDF5)
 
   # Check to see if HDF5 library is 1.10.6 or greater.
   # Used to control path name conversion
-  IF(${HDF5_VERSION} VERSION_LESS "1.10.6")
-    SET(HDF5_UTF8_PATHS FALSE)
-  ELSE()
-    SET(HDF5_UTF8_PATHS TRUE)
-  ENDIF()
+  # Check HDF5 version
+  SET(HDF5_UTF8_PATHS OFF)
+  CHECK_C_SOURCE_COMPILES("
+  #include <H5public.h>
+  int main() {
+#if (H5_VERS_MAJOR*10000 + H5_VERS_MINOR*100 + H5_VERS_RELEASE < 11006)
+      choke me;
+#endif
+  }" HDF5_VERSION_1106)
+
+  IF(HDF5_VERSION_1106)
+   SET(HDF5_UTF8_PATHS ON)
+  ENDIF(HDF5_VERSION_1106)
+
   MESSAGE("-- Checking for HDF5 version 1.10.6 or later: ${HDF5_UTF8_PATHS}")
 
   SET(H5_USE_16_API 1)
@@ -869,18 +878,6 @@ IF(USE_HDF5)
   #option to include HDF5 High Level header file (hdf5_hl.h) in case we are not doing a make install
   INCLUDE_DIRECTORIES(${HDF5_HL_INCLUDE_DIR})
 
-  # Check HDF5 version
-  CHECK_C_SOURCE_COMPILES("
-  #include <H5public.h>
-  int main() {
-#if (H5_VERS_MAJOR*10000 + H5_VERS_MINOR*100 + H5_VERS_RELEASE < 11006)
-      choke me;
-#endif
-  }" HDF5_VERSION_1106)
-
-  IF(HDF5_VERSION_1106)
-   SET(HDF5_UTF8_PATHS ON)
-  ENDIF()
 
 ENDIF(USE_HDF5)
 
@@ -2240,7 +2237,13 @@ configure_file(
   ${netCDF_SOURCE_DIR}/include/netcdf_dispatch.h.in
   ${netCDF_BINARY_DIR}/include/netcdf_dispatch.h @ONLY NEWLINE_STYLE LF)
 
-#####
+# Used in the `configure_file` calls below
+SET(ISCMAKE "1")
+IF(MSVC)
+SET(ISMSVC "1")
+ENDIF()
+
+####
 # Build test_common.sh
 #####
 SET(EXTRA_DIST ${EXTRA_DIST} ${CMAKE_CURRENT_SOURCE_DIR}/test_common.in)
@@ -2249,15 +2252,9 @@ SET(TOPBUILDDIR "${CMAKE_CURRENT_BINARY_DIR}")
 SET(ISMSVC "${MSVC}")
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/test_common.in ${CMAKE_CURRENT_BINARY_DIR}/test_common.sh @ONLY NEWLINE_STYLE LF)
 
-
 #####
 # Build nc_test4/findplugin.sh
 #####
-SET(ISCMAKE "1")
-IF(MSVC)
-SET(ISMSVC "1")
-ENDIF()
-
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/nc_test4/findplugin.in ${CMAKE_CURRENT_BINARY_DIR}/nc_test4/findplugin.sh @ONLY NEWLINE_STYLE LF)
 
 IF(ENABLE_EXAMPLES)
