--- dap4_test/Makefile.in
+++ dap4_test/Makefile.in
@@ -693,6 +693,9 @@
              tst_curlopt.sh d4test_common.sh \
 	     daptestfiles dmrtestfiles cdltestfiles nctestfiles misctestfiles \
 	     baseline baselineraw baselineremote baselinehyrax baselinethredds CMakeLists.txt test_common.h
+# Ensure no arguments are passed to test_fillmismatch.sh
+$(srcdir)/test_fillmismatch.sh:
+    $(RUNSH) $(srcdir)/test_fillmismatch.sh
 
 # This should only be left behind if using parallel io
 CLEANFILES = *.exe tmp_* \
