diff --color -Naur netcdf-c-4.9.0.orig/CMakeLists.txt netcdf-c-4.9.0/CMakeLists.txt
--- netcdf-c-4.9.0.orig/CMakeLists.txt	2022-06-10 18:04:15.000000000 -0300
+++ netcdf-c-4.9.0/CMakeLists.txt	2022-06-13 10:29:40.177921583 -0300
@@ -893,11 +893,19 @@
   # Check to see if HDF5 library is 1.10.6 or greater.
   # Used to control path name conversion
   ####
-  IF(${HDF5_VERSION} VERSION_GREATER "1.10.5")
-    SET(HDF5_UTF8_PATHS ON)
-  ELSE()
-    SET(HDF5_UTF8_PATHS OFF)
-  ENDIF()
+  # Check HDF5 version
+  SET(HDF5_UTF8_PATHS OFF)
+  CHECK_C_SOURCE_COMPILES("
+  #include <H5public.h>
+  int main() {
+#if (H5_VERS_MAJOR*10000 + H5_VERS_MINOR*100 + H5_VERS_RELEASE < 11006)
+      choke me;
+#endif
+  }" HDF5_VERSION_1106)
+
+  IF(HDF5_VERSION_1106)
+   SET(HDF5_UTF8_PATHS ON)
+  ENDIF(HDF5_VERSION_1106)
 
   MESSAGE("-- HDF5_UTF8_PATHS (HDF5 version 1.10.6+): ${HDF5_UTF8_PATHS}")
 
diff --color -Naur netcdf-c-4.9.0.orig/nc_test4/CMakeLists.txt netcdf-c-4.9.0/nc_test4/CMakeLists.txt
--- netcdf-c-4.9.0.orig/nc_test4/CMakeLists.txt	2022-06-10 18:04:15.000000000 -0300
+++ netcdf-c-4.9.0/nc_test4/CMakeLists.txt	2022-06-13 10:29:54.401820897 -0300
@@ -59,10 +59,6 @@
 
 ENDIF(BUILD_UTILITIES)
 
-IF(${HDF5_VERSION} VERSION_GREATER "1.10.0")
-  SET(NC4_TESTS ${NC4_TESTS} tst_virtual_datasets)
-ENDIF(${HDF5_VERSION} VERSION_GREATER "1.10.0")
-
 ##
 # The shell script, run_empty_vlen_test.sh,
 # depends on the 'tst_empty_vlen_unlim' binary.
