--- netcdf-c.orig/CMakeLists.txt
+++ netcdf-c/CMakeLists.txt
@@ -831,11 +831,7 @@ IF(USE_HDF5)
 
   # Check to see if HDF5 library is 1.10.6 or greater.
   # Used to control path name conversion
-  IF(${HDF5_VERSION} VERSION_LESS "1.10.6")
-    SET(HDF5_UTF8_PATHS FALSE)
-  ELSE()
-    SET(HDF5_UTF8_PATHS TRUE)
-  ENDIF()
+  SET(HDF5_UTF8_PATHS TRUE)
   MESSAGE("-- Checking for HDF5 version 1.10.6 or later: ${HDF5_UTF8_PATHS}")
 
   SET(H5_USE_16_API 1)
