diff -ru netcdf-c-4.9.2.orig/CMakeLists.txt netcdf-c-4.9.2/CMakeLists.txt
--- netcdf-c-4.9.2.orig/CMakeLists.txt	2025-03-04 14:14:09.494260204 +0100
+++ netcdf-c-4.9.2/CMakeLists.txt	2025-03-04 14:14:21.989263798 +0100
@@ -2420,7 +2420,7 @@
 # Subdirectory CMakeLists.txt files should specify their own
 # 'install' files.
 # Including 'CPack' kicks everything off.
-INCLUDE(InstallRequiredSystemLibraries)
+# INCLUDE(InstallRequiredSystemLibraries)
 CONFIGURE_FILE(
   ${CMAKE_CURRENT_SOURCE_DIR}/FixBundle.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/FixBundle.cmake
